### DISCUSSION TITLE
North indicator / bitmap at fixed screen position?

### DISCUSSION THREAD

**Discussion by moving-bits** [mapsforge/vtm#929] [Q&A] [answered]
I wand to add a north indicator to the map, being displayed in top-left corner. The icon shall rotate according to map rotation, but otherwise stay fixed at that position.

I'm able to position a marker at a certain geo position using an `ItemizedLayer`, but as soon as I rotate the map, the marker will move out of sight (when its coordinates are no longer visible).

Is there a way to detect the geo location for a fixed screen position (e. g. at 10% map view width from left / 10% map view height from top), taken map rotation into account? Then I could update the marker's geo location.

**Comment by moving-bits** [comment, marked as answer]
Hm, could probably answer this myself: Create a view on top of the map view, no map layer necessary...

(Sometimes you need to write things down to get the right idea... Putting an `ImageView` on top of the mapView does work like a charm.)

**Comment by devemux86** [comment]
The proper way to do such external additions (as you found)
is with extra views over the map view (like the zoom buttons).

**Comment by jocieldo** [comment]
Hey guys, I'm trying to do the same, the item is already in place, but how I rotate it? What is the event from the map I need to listen ? I tried the map events to the the rotate_event but it only triggers when I rotate the map manually, when I rotate by code, nothing is triggered. Can you help me on this? Thanks

**Reply by devemux86** [reply]
To listen for map events, see [MapScaleBarLayer](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/scalebar/MapScaleBarLayer.java) and [PathOverlayActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/PathOverlayActivity.java).

```java
map.events.bind(new UpdateListener() {
    @Override
    public void onMapEvent(Event e, MapPosition mapPosition) {
        if (e == Map.UPDATE_EVENT)
            return;
        ...
    }
});
```

**Reply by jocieldo** [reply]
That's awesome.. I was trying to figure out a event for that, and the solution was just put there without any event and it's running very smoothly. LOL

Thank you so much!

```java
    override fun onMapEvent(e: Event?, mapPosition: MapPosition?) {
        when (e) {
            Map.ANIM_END -> {
                ...
            }
        }
        binding.btnNorth.rotation = mapView.map().mapPosition.bearing   // No fixed event 
    }
```
